core/src/components/button/button.ionic.scss

```diff
@@ -53,45 +53,59 @@
   --ripple-color: var(--background-activated);
 }
 
+:host(.button-solid.ion-color) {
+  --background-activated: #{globals.current-color(shade)};
+  --background-hover: #{globals.current-color(shade)};
+  --background: #{globals.current-color(base)};
+  --color: #{globals.current-color(contrast)};
+}
+
```

Comment:
This code is not needed since it's already implemented: https://github.com/ionic-team/ionic-framework/blob/21293da293a6d7d1d846be03a98eb9a20d7cc3f7/core/src/components/button/button.ionic.scss#L224-L226 and https://github.com/ionic-team/ionic-framework/blob/21293da293a6d7d1d846be03a98eb9a20d7cc3f7/core/src/components/button/button.ionic.scss#L236-L240


```suggestion
```